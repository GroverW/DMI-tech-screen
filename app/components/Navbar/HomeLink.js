/**
 * A generic anchor tag
 */

import styled from 'styled-components';
import { Link } from 'react-router-dom';

const HomeLink = styled(Link)`
  padding: 5px 10px;
  margin-right: auto;
  color: ${props => props.theme.font};
  text-decoration: none;
`;

export default HomeLink;
