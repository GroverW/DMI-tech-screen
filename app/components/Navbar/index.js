/**
 *
 * Navbar
 *
 */

import React from 'react';
import { NavLink } from 'react-router-dom';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function Navbar() {
  return (
    <nav className="Navbar">
      <NavLink exact to="/">
        <FormattedMessage {...messages.home} />
      </NavLink>
      <NavLink exact to="/strings">
        <FormattedMessage {...messages.viewAll} />
      </NavLink>
      <NavLink exact to="/strings/new">
        <FormattedMessage {...messages.addNew} />
      </NavLink>
    </nav>
  );
}

Navbar.propTypes = {};

export default Navbar;
