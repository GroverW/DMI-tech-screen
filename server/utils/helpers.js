const { RESULTS_PER_PAGE } = require('../config');

const getPageStart = page =>
  page && Number.isInteger(+page)
    ? Math.max(page - 1, 0) * RESULTS_PER_PAGE
    : 0;

module.exports = {
  getPageStart,
};
