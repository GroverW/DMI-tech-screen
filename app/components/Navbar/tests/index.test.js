/**
 *
 * Tests for Navbar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';
import { renderWithRouter } from '../../../tests/helpers/testHelper';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Navbar from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Navbar />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderWithRouter(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Navbar />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderWithRouter(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Navbar />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
