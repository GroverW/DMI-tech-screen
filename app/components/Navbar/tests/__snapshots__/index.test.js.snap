// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Should render and match the snapshot 1`] = `
<div
  class="Nav-x8yy12-0 byJLoV"
>
  <a
    class="HomeLink-sc-19918yo-0 glzkdw"
    href="/"
    theme="[object Object]"
  >
    <span>
      String Central
    </span>
  </a>
  <a
    class="HeaderLink-sc-16cp3q6-0 clJWfa"
    href="/all-strings"
    theme="[object Object]"
  >
    <span>
      View All Strings
    </span>
  </a>
  <a
    class="HeaderLink-sc-16cp3q6-0 clJWfa"
    href="/all-strings/new"
    theme="[object Object]"
  >
    <span>
      Add New String
    </span>
  </a>
  <div>
    <select
      class="Toggle-oqiqhq-0 jVmmhU"
    >
      <option
        id="light"
        value="light"
      >
        light
      </option>
      <option
        id="dark"
        value="dark"
      >
        dark
      </option>
    </select>
  </div>
</div>
`;
